src/unix/event.rs

```diff
@@ -0,0 +1,346 @@
+    use std::{ffi::CString, io::pipe, os::fd::IntoRawFd, process::Command};
+
+    #[test]
+    fn test_notify_and_consume() {
+        let (reader, writer) = pipe().expect("Failed to create pipe");
+        // SAFETY: Safe because we check the fd is valid. And the kernel gave us an fd that we own.
+        let notifier = unsafe { EventNotifier::from_raw_fd(writer.into_raw_fd()) };
+        // SAFETY: Safe because we check the fd is valid. And the kernel gave us an fd that we own.
+        let consumer = unsafe { EventConsumer::from_raw_fd(reader.into_raw_fd()) };
+
+        notifier.notify().unwrap();
+        assert!(consumer.consume().is_ok());
+    }
+
+    #[test]
+    fn test_clone() {
+        let (reader, writer) = pipe().expect("Failed to create pipe");
+        // SAFETY: Safe because we check the fd is valid. And the kernel gave us an fd that we own.
+        let notifier = unsafe { EventNotifier::from_raw_fd(writer.into_raw_fd()) };
+        // SAFETY: Safe because we check the fd is valid. And the kernel gave us an fd that we own.
+        let consumer = unsafe { EventConsumer::from_raw_fd(reader.into_raw_fd()) };
+
+        let cloned_notifier = notifier.try_clone().expect("Failed to clone notifier");
+        let cloned_consumer = consumer.try_clone().expect("Failed to clone consumer");
+
+        cloned_notifier.notify().unwrap();
+        assert!(cloned_consumer.consume().is_ok());
+    }
+
+    #[test]
+    fn test_new_event_notifier_and_consumer() {
+        let (consumer, notifier) = new_event_consumer_and_notifier(EventFlag::empty())
+            .expect("Failed to create notifier and consumer");
+        notifier.notify().unwrap();
+        assert!(consumer.consume().is_ok());
+    }
+
+    #[test]
+    fn test_read_nothing() {
+        let (consumer, _notifier) = new_event_consumer_and_notifier(EventFlag::NONBLOCK)
+            .expect("Failed to create notifier and consumer");
+        let r = consumer.consume();
+        match r {
+            Err(ref inner) if inner.kind() == io::ErrorKind::WouldBlock => (),
+            _ => panic!("Unexpected"),
+        }
+    }
+
+    #[test]
+    fn test_cloexec() {
+        let (_consumer, _notifier) = new_event_consumer_and_notifier(EventFlag::CLOEXEC)
+            .expect("Failed to create notifier and consumer");
+
+        // SAFETY: This is safe because we are checking the return value is valid.
+        unsafe {
+            let pid = libc::fork();
+            if pid == 0 {
+                let path = CString::new("/bin/sleep").expect("CString::new failed");
+                let arg0 = CString::new("sleep").expect("CString::new failed");
+                let arg1 = CString::new("1").expect("CString::new failed");
+                let ret = libc::execl(
+                    path.as_ptr(),
+                    arg0.as_ptr(),
+                    arg1.as_ptr(),
+                    std::ptr::null::<i8>(),
+                );
+                assert_eq!(ret, 0, "execl failed");
```

Comment:
Can you explain a bit this test, for example, why you need to call `/bin/sleep` ?

Reply:
okay, IIUC it's just something to pass to `execl`, but how we are sure the child called `execl` before the parent calls `lsof` ?

Reply:
https://github.com/rust-vmm/vmm-sys-util/pull/244/files#diff-b817124a7a414d687ff2826960adbb7bf304a2770b0f6ee70b3daa5ca4a16113R329 Make sure the process exec sleep through monitoring the cmdline of the process